### PR TITLE
gambit-scheme 4.9.0

### PR DESCRIPTION
**Pull request by ober**
- [X ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
```
gambit-scheme:
  * Non-libraries were installed to "/usr/local/opt/gambit-scheme/lib"
    Installing non-libraries to "lib" is discouraged.
```

File: Formula/gambit-scheme.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
+class GambitScheme < Formula
+  desc "Gambit Scheme"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Needs a better description

File: Formula/gambit-scheme.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  homepage "http://gambitscheme.org"
+  url "https://github.com/gambit/gambit.git", :tag => "v4.9.0"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Needs associated `:revision`. And we prefer to build from github tarball if possible.

File: Formula/gambit-scheme.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  depends_on "gcc@6"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: This needs to build with latest gcc (`depends_on "gcc"`)  to be considered for inclusion in Homebrew core.

Comment (fxcoudert) [reply Formula/gambit-scheme.rb]: We need a comment stating why this cannot compile with `clang`. And then `fails_with :clang`.

File: Formula/gambit-scheme.rb [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  def install
+    args = %W[--prefix=#{prefix}]
+
+    ENV["CC"] = Formula["gcc@6"].bin/"gcc-6"
+
+    if build.with? "disable-ssl-verification"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: No options. Provide a "best use case" version that will have most features and be useful for as many users as possible.

File: Formula/gambit-scheme.rb [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    if build.with? "gerbil-options"
+      args << "--enable-default-runtime-options=f8,-8,t8"
+    end
+
+    if build.with? "poll"
+      args << "--enable-poll"
+    end
+
+    if build.with? "openssl"
+      depends_on "openssl"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: `depends_on` goes at the toplevel

File: Formula/gambit-scheme.rb [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+      args << "--enable-openssl"
+    end
+
+    system "./configure", *args
+    system "make"
+    ENV.deparallelize
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Why? Report this as a bug to upstream.

File: Formula/gambit-scheme.rb [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+    bin.install "gsc/gsc"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: No `make install`? We definitely do not want to install everything by hand

File: Formula/gambit-scheme.rb [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        gambit#.scm
+        libgambitgsc.a
+        libgambitgsi.a
+        r5rs#.scm
+        syntax-case.scm
+        _asm#.scm ].each do |l|
+      lib.install Dir["lib/#{l}"]
+    end
+
+    bin.install_symlink "bin/gsc" => "gsc-script"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: If upstream does not install that symlink, why should we?

File: Formula/gambit-scheme.rb [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    prefix.install_symlink prefix => "#{prefix}/v#{version}"
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Definitely a bad idea.

File: Formula/gambit-scheme.rb [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+  end
+
+  test do
+    output = `#{bin}/gsi -e "(for-each write '(0 1 2 3 4 5 6 7 8 9))"`
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Please use `shell_output`. Otherwise it's a good test!

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/gambit-scheme.rb (hunk 5 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: except if it's optional.

File: Formula/gambit-scheme.rb (hunk 6 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: Bug? Please explain.

File: Formula/gambit-scheme.rb (hunk 5 above)
Comment (fxcoudert) [reply Formula/gambit-scheme.rb]: No. `depends_on` never goes inside the `install` block.

File: Formula/gambit-scheme.rb (hunk 6 above)
Comment (fxcoudert) [reply Formula/gambit-scheme.rb]: If it doesn't build in parallel, it's a makefile bug. Please report it.

File: Formula/gambit-scheme.rb (hunk 5 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: How do we handle optional dependencies as that was triggering audit.

File: Formula/gambit-scheme.rb (hunk 6 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: disable parallel for make install

**Pull request closed without merging**